src/Workspaces/SharedUtilitiesAndExtensions/Workspace/CSharp/LanguageServices/CSharpAddImportsService.cs

```diff
@@ -29,6 +27,13 @@ public CSharpAddImportsService()
         {
         }
 
+#if !CODE_STYLE
+        public override bool PlaceImportsInsideNamespaces(Options.OptionSet optionSet)
+        {
+            return optionSet.GetOption(CSharpCodeStyleOptions.PreferredUsingDirectivePlacement).Value == AddImportPlacement.InsideNamespace;
+        }
+#endif
```

Comment:
intereting.  how does this work with code style?

Reply:
Not sure what do you mean, could you elaborate?

Reply:
yeah... i'm def curious what's going on here.  the layering seems strange.

Reply:
Filed https://github.com/dotnet/roslyn/issues/59179